scripts/machine-launch-script.sh

```diff
@@ -1,57 +1,156 @@
 #!/bin/bash
 
+LAUNCH_STATUS_PATH="/home/centos"
+
+usage()
+{
+    echo "Usage: $0 [options]"
+    echo
+    echo "[-help]                  List this help"
+    echo "[-launch-status-path]    Path to store machine-launchstatus logs. Default: $LAUNCH_STATUS_PATH/<HERE>"
+}
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+        -help)
+            usage
+            exit 1
+            ;;
+        -launch-status-path)
+            shift
+            LAUNCH_STATUS_PATH=$1
+            shift
+            ;;
+        *)
+            echo "Invalid Argument: $1"
+            usage
+            exit 1
+            ;;
+    esac
+done
+
 set -ex
 set -o pipefail
 
-echo "machine launch script started" > /home/centos/machine-launchstatus
-sudo chgrp centos /home/centos/machine-launchstatus
-sudo chown centos /home/centos/machine-launchstatus
+FLAVOR=`grep '^ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+VERSION=`grep '^VERSION_ID=' /etc/os-release | awk -F= '{print $2}' | tr -d '"'`
+MAJOR=${VERSION%.*}
+ARCH=`uname -m`
 
-{
-sudo yum install -y ca-certificates
-sudo yum install -y mosh
-sudo yum groupinstall -y "Development tools"
-sudo yum install -y gmp-devel mpfr-devel libmpc-devel zlib-devel vim git java java-devel
-curl https://www.scala-sbt.org/sbt-rpm.repo | sudo tee /etc/yum.repos.d/scala-sbt-rpm.repo
-sudo yum install -y sbt texinfo gengetopt libffi-devel
-sudo yum install -y expat-devel libusb1-devel ncurses-devel cmake "perl(ExtUtils::MakeMaker)"
-# deps for poky
-sudo yum install -y python36 patch diffstat texi2html texinfo subversion chrpath git wget
-# deps for qemu
-sudo yum install -y gtk3-devel
-# deps for firesim-software (note that rsync is installed but too old)
-sudo yum install -y python36-pip python36-devel rsync
-# Install GNU make 4.x (needed to cross-compile glibc 2.28+)
-sudo yum install -y centos-release-scl
-sudo yum install -y devtoolset-8-make
-
-# install DTC
-sudo yum -y install dtc
-
-# get a proper version of git
-sudo yum -y remove git
-sudo yum -y install epel-release
-sudo yum -y install https://repo.ius.io/ius-release-el7.rpm
-sudo yum -y install git224
+if [ $FLAVOR != "ubuntu" ] && [ $FLAVOR != "centos" ]; then
+    echo "Unknown OS flavor $FLAVOR"
+    exit 1
+fi
 
-# install verilator
-if ! command -v verilator &> /dev/null
-then
-    git clone http://git.veripool.org/git/verilator
-    cd verilator/
-    git checkout v4.034
-    autoconf && ./configure && make -j4 && sudo make install
-    cd ..
+echo "machine launch script started" > $LAUNCH_STATUS_PATH/machine-launchstatus
+if [ $(whoami) == "centos" ]; then
+    sudo chgrp centos $LAUNCH_STATUS_PATH/machine-launchstatus
+    sudo chown centos $LAUNCH_STATUS_PATH/machine-launchstatus
```

Comment:
Abe, are you sure that when the script is copy-pasted into userdata that it gets run as centos user?  I think that it gets run as root and that's the reason why they chgrp and chown it.  Probably makes more sense to just `chmod ugo+r` the log.

Reply:
It does run as root. The lines added were to make sure the `centos` user had access to the log. I'm happy to use `ugo+r` instead.

Reply:
I think my point was that the whoami wouldn't return centos when you wanted to chown the log. Also centos won't exist on Ubuntu. Should just make sure it is world readable